sklearn/decomposition/_dict_learning.py

```diff
@@ -2086,11 +1835,9 @@ class MiniBatchDictionaryLearning(_BaseSparseCoding, BaseEstimator):
     _parameter_constraints: dict = {
         "n_components": [Interval(Integral, 1, None, closed="left"), None],
         "alpha": [Interval(Real, 0, None, closed="left")],
-        "n_iter": [
-            Interval(Integral, 0, None, closed="left"),
-            Hidden(StrOptions({"deprecated"})),
-        ],
-        "max_iter": [Interval(Integral, 0, None, closed="left"), None],
+        # `max_iter=None` is equivalent to 1_000 and was used during n_iter deprecation
+        # Hide it to not silently support it.
+        "max_iter": [Interval(Integral, 0, None, closed="left"), Hidden(None)],
```

Comment:
we should deprecate max_iter=None now, don't you think ?

Reply:
This is something that I was wondering. Maybe we can do that because we should not have anyone that explicitly set to `None`. It should not be too annoying.